lib/resty/core/shdict.lua

```diff
@@ -387,6 +401,25 @@ local function shdict_flush_all(zone)
 end
 
 
+local function shdict_get_used_bytes(zone)
+    zone = check_zone(zone)
+
+    local slot_count_buf = ffi_new("size_t[1]")
+    C.ngx_http_lua_ffi_shdict_get_stats_slot_count(zone, slot_count_buf)
+    local stats = ffi_new("ngx_slab_stat_t[?]", slot_count_buf[0])
```

Comment:
Similar to new Lua table creation, we should also avoid creating new cdata objects on hot code paths.